services/navigation/builtin/builtin.go

```diff
@@ -340,6 +338,13 @@ func (svc *builtIn) AddWaypoint(ctx context.Context, point *geo.Point, extra map
 }
 
 func (svc *builtIn) RemoveWaypoint(ctx context.Context, id primitive.ObjectID, extra map[string]interface{}) error {
+	goalWaypoint, err := svc.nextWaypoint(ctx)
+	if err != nil {
+		return err
+	}
+	if goalWaypoint.ID == id {
+		svc.cancelFunc()
```

Comment:
I think this needs a mutex.

If multiple clients are calling this service, then there is the potential for one client to cause cancelFunc to be called, while another in causing it to be replaced.

Reply:
Also, this will need to be replaced after being called. Once `cancelFunc()` has been called, the context will remain cancelled and `startWaypoint` will be unable to function until there is a new valid `svc.cancelCtx`

Reply:
Or, more specifically; is the goal here that a removal of the current waypoint will result in the cessation of all movement until `SetMode` is called manually? Or are we intended to instead plan to the next waypoint that would be in the list?

Currently the behavior is the former. Based on RSDK-3299, it seems like it should be the latter instead.